src/components/ColorPickerContainer.ts

```diff
@@ -0,0 +1,286 @@
+                this.props.mxObject.set(this.props.colorAttribute, newColor);
+                this.handleOnChange();
+            }
+            this.setState(state);
+        } else {
+            this.setState({ alertMessage: "Invalid color", color: event.target.value });
+        }
+    }
+
+    private handleOnChange = () => {
+        const { mxform, mxObject, onChangeEvent, onChangeMicroflow, onChangeNanoflow, onChangePage } = this.props;
+        if (mxObject && ((this.previousColor !== this.state.color) || this.colorChanged)) {
+            const context = new mendix.lib.MxContext();
+            context.setContext(mxObject.getEntity(), mxObject.getGuid());
+            if (onChangeEvent === "callMicroflow" && onChangeMicroflow) {
+                window.mx.ui.action(onChangeMicroflow, {
+                    error: error => window.mx.ui.error(`Error while executing microflow ${onChangeMicroflow}: ${error.message}`), // tslint:disable-line max-line-length
+                    params: {
+                        applyto: "selection",
+                        guids: [ mxObject.getGuid() ]
+                    },
+                    origin: mxform
+                });
+            } else if (onChangeEvent === "showPage" && onChangePage) {
+                window.mx.ui.openForm(onChangePage, {
+                    context,
+                    error: error => window.mx.ui.error(`Error while opening page ${onChangePage}: ${error.message}`),
+                    location: this.props.openPageLocation
+                });
+            } else if (onChangeEvent === "callNanoflow" && onChangeNanoflow.nanoflow) {
+                window.mx.data.callNanoflow({
+                    context,
+                    error: error => window.mx.ui.error(`Error while executing the onchange nanoflow: ${error.message}`),
+                    nanoflow: onChangeNanoflow,
+                    origin: mxform
+                });
+            }
+        }
+    }
+
+    public static validateProps(props: ColorPickerContainerProps): string {
+        const message: string[] = [];
+        if (props.onChangeEvent === "callMicroflow" && !props.onChangeMicroflow) {
+            message.push("On change event is set to 'Call a microflow' but no microflow is selected");
+        } else if (props.onChangeEvent === "showPage" && !props.onChangePage) {
+            message.push("On change event is set to 'Show a page' but no page is selected");
+        } else if (props.onChangeEvent === "callNanoflow" && (JSON.stringify(props.onChangeNanoflow) === JSON.stringify({}))) {
+            message.push("On change event is set to 'Call a nanoflow' but no nanoflow is selected");
+        }
+        if (props.label.trim() && props.labelWidth > 11) {
+            message.push("Label width (weight) should not be greater than 11");
+        }
+
+        return message.length ? `Configuration error in widget ${props.friendlyId}: ${message.join(", ")}` : "";
```

Comment:
we should no longer use in the us the friendelyId. because it friendly to the developer but not to de end user.
Though we still should log an error to the console with the friendlyId. As from 7.14 the console log is send to the server. So when there is config issue in the production the developer can still find error by watching the log.

Though be carefull that we only log the logger.error when the it is available (runtime only, not web modeler)

Reply:
So we should do it for the developer but at the time they are building the app am sure they have to do some testing...maybe this can help them